src/ExceptionHandler.f90

```diff
@@ -366,18 +442,22 @@ PURE SUBROUTINE initCounter(e)
 !>
     PURE SUBROUTINE reset(e)
       CLASS(ExceptionHandlerType),INTENT(INOUT) :: e
+      INTEGER(SIK) :: i
       NULLIFY(e%surrogate)
-      e%nInfo=0
-      e%nWarn=0
-      e%nError=0
-      e%nFatal=0
-      e%nDebug=0
       e%lastMesg=''
       e%logFileUnit=666
       e%stopOnError=.TRUE.
       e%logFileActive=.FALSE.
       e%quiet=(/.FALSE.,.FALSE.,.TRUE.,.FALSE./)
       e%verbose=(/.TRUE.,.TRUE.,.FALSE.,.TRUE./)
+      IF(ALLOCATED(e%exceptionRegistry)) THEN
+        DO i=1,SIZE(e%exceptionRegistry)-1
```

Comment:
I want to hazard a guess and say it's `SIZE()-1` because we never want to change the `EXCEPTION_FATAL_ERROR` quiet mode and verbosity, since the `FATAL_ERROR` parameter value is 5 and the number of exceptions in the registry is also 5.  Is that the case?  If so, it should really check if the `%exceptionRegistry` is the `FATAL_ERROR` and skip, right?

Reply:
Dan, good point.  That's probably why.

This does raise a design-related question though: if we're generalizing the exception classes, should we have one that is always handled differently with regard to verbosity?  If so, we should probably allow any of the exceptions to be controlled similarly.  For example, maybe when the exception gets initialized, it has an `immutable` attribute that is false by default, but can be set to true for things like fatal errors.

This probably doesn't need to be decided on now, but wanted to raise the question

Reply:
I think an `immutable` attribute makes the most sense, and was thinking along the same lines.